ravenframework/Optimizers/parentSelectors/parentSelectors.py

```diff
@@ -62,91 +64,141 @@ def rouletteWheel(population,**kwargs):
     roulettePointer = randomUtils.random(dim=1, samples=1)
     # initialize Probability
     counter = 0
-    if np.all(fitness.data>=0) or np.all(fitness.data<=0):
-      selectionProb = fitness.data/np.sum(fitness.data) # Share of the pie (rouletteWheel)
+    if np.all(fitness>=0) or np.all(fitness<=0):
+      selectionProb = fitness/np.sum(fitness) # Share of the pie (rouletteWheel)
```

Comment:
Is it possible that np.sum(fitness) == 0 here? If so, this need to be caught.

Reply:
Probably unlikely, but I did add a try/except block for (ZeroDivisionError, RuntimeWarning).